test/spec/OnLocalRemoteChangeSpec.js

```diff
@@ -60,11 +58,40 @@ describe("OnLocal/RemoteChange", function() {
         // wait for observer and mocked initial HTTP response
         setTimeout(function() {
             expect(puppet.obj.hello).toEqual("onRemoteChange callback");
-            expect(receivedSpy.calls.mostRecent().args[0]).toEqual([{
+            var mostRecentCall = receivedSpy.calls.mostRecent();
+            expect(mostRecentCall.args[0]).toEqual([{
                 "op": "replace",
                 "path": "/hello",
                 "value": "onRemoteChange callback"
             }]);
+            expect(mostRecentCall.args[1]).toEqual([{removed: 'world'}]);
+            done();
+        }, 100); // for FF
+    });
+
+    it("should fire patch-applied event for applied patches", function(done) {
```

Comment:
Could you also add a test  for `should NOT fire patch-applied event for received, and queued patches`?